src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,71 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import java.util.Objects;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        verifyUserForNull(user);
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+        verifyUserExist(user);
+        return storageDao.add(user);
+    }
+
+    private void verifyUserForNull(User user) {
+        if (user == null) {
+            throw new NonValidUserDataException(NonValidUserDataException.class.getName()
+                    + "User is null");
```

Comment:
Why do we need  `NonValidUserDataException.class.getName() info` ? This will be present in stack trace.
```suggestion
            throw new NonValidUserDataException("User is null");
```
Fix all places in the code please